snakemake/remote/__init__.py

```diff
@@ -294,3 +295,46 @@ def local_path(self):
     @property
     def remote_path(self):
         return self.path_remainder
+
+
+class AutoRemoteProvider:
+    def __init__(self, *args, **kwargs):
+        self.constructor_args = args
+        self.constructor_kwargs = kwargs
+
+    @property
+    def protocol_mapping(self):
+        # TODO: how to include all providers automatically (AbstractRemoteProvider.__subclasses__)?
```

Comment:
Sounds reasonable, can you try that out? We might need to add a static field to each remote provider that tells whether it is eligible for being automated like this (some might require special args).

Reply:
I implemented one possible, though maybe not robust, way of gathering all implemented RemoteProviders.
We could add a new argument to `AutoRemoteProvider.remote` which allows passing special args to each instance.